presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergResourceFactory.java

```diff
@@ -128,7 +131,10 @@ private String getCatalogCacheKey(ConnectorSession session)
     private Configuration getHadoopConfiguration(ConnectorSession session)
     {
         Configuration configuration = new Configuration(false);
+
         if (hadoopConfigResources.isEmpty()) {
+            s3ConfigurationUpdater.updateConfiguration(configuration);
```

Comment:
Thank you @hmadison for fixing up support for S3-backed Iceberg tables.

I have a question, Is there any reason we are not updating S3 configs always?
I mean we can also have the scenario when we have an iceberg table backed by an S3 filesystem and we want to supply some extra Hadoop config-resource in the catalog, then in that case how s3 configs would be updated?





[11:26](https://ibm-analytics.slack.com/archives/D05HQ5MFJBW/p1693331799970769)
So I think we shouldn't keep it under the conditional check of hadoop config resource?